### PR TITLE
Delete rogue registration requests without sending a reason

### PR DESCRIPTION
**Pull request by garaimanoj**

### CODE REVIEW

**Comment by garaimanoj** [comment]
image

**Comment by rmiccoli** [comment]
> image

In my opinion, when the box "Do not send email to the rejected user" is checked, the reason shouldn't be mandatory. Actually it becomes quite useless because it is neither logged nor saved to the database.

File: iam-login-service/src/main/java/it/infn/mw/iam/registration/DefaultRegistrationRequestService.java [hunk 1]
```diff
@@ -358,9 +358,12 @@ private RegistrationRequestDto handleConfirm(IamRegistrationRequest request) {
...
 
   private RegistrationRequestDto handleReject(IamRegistrationRequest request,
-      Optional<String> motivation) {
+      Optional<String> motivation, boolean doNotSendEmail) {
     request.setStatus(REJECTED);
-    notificationFactory.createRequestRejectedMessage(request, motivation);
+    if(!doNotSendEmail){
+      notificationFactory.createRequestRejectedMessage(request, motivation);
+    }
+
```
Comment (enricovianello) [inline iam-login-service/src/main/java/it/infn/mw/iam/registration/DefaultRegistrationRequestService.java:366]: I would add the `motivation` to the message published into RegistrationRejectEvent (line 371), knowing it's still required

**Review by enricovianello — CHANGES_REQUESTED** [review]
Easy fix: I would add the motivation to the message published into RegistrationRejectEvent

File: iam-login-service/src/main/java/it/infn/mw/iam/registration/DefaultRegistrationRequestService.java (hunk 1 above)
Comment (garaimanoj) [reply iam-login-service/src/main/java/it/infn/mw/iam/registration/DefaultRegistrationRequestService.java:366]: Done

**Review by rmiccoli — APPROVED** [review]

**Review by enricovianello — APPROVED** [review]

**Pull request merged**